q2_feature_table/tests/filter/test_filter_samples.py

```diff
@@ -799,14 +799,31 @@ def test_combine_exclude_ids_and_features_filters(self):
                       ['O1', 'O2'],
                       ['S1', 'S2', 'S3'])
         where = "SampleType='tongue'"
-        actual = filter_samples(table,
-                                metadata=metadata,
-                                where=where,
-                                exclude_ids=True,
-                                min_features=1,
-                                max_features=1)
+        with self.assertRaisesRegex(ValueError, 'empty'):
+            filter_samples(table,
+                           metadata=metadata,
+                           where=where,
+                           exclude_ids=True,
+                           min_features=1,
+                           max_features=1)
+
+    def test_allow_empty_table_true(self):
```

Comment:
This is great! For the rest of your tests, I'd make sure and add a test case like this one that asserts an empty table is allowed (for each method where this is used).

Reply:
Just to clarify here: Each test where I filter all the samples and test raising the 'Empty table' error should also have a corresponding test where the empty table is accepted with the allow_empty_table=True. Is that right?

Reply:
Yes, that's correct! You should be testing that both the true and false cases work as expected for each method where this new parameter is used.